examples/src/main/java/io/grpc/examples/helloworld/HelloWorldServer.java

```diff
@@ -0,0 +1,217 @@
+          return false;
+        }
+      } else {
+        try {
+          zoo.setData(server_addr, current_time.getBytes(), stat.getVersion());
+        } catch (Exception e) {
+          logger.severe("Failed to update server_data");
+          return false;
+        }
+      }
+    } catch (Exception e) {
+      logger.severe("Failed to add server_data");
+      return false;
+    }
+    return true;
+  }
+
+  // Method to disconnect from zookeeper server
+  public void close() throws InterruptedException {
+    zoo.close();
+  }
+}
+
+/**
+ * Server that manages startup/shutdown of a {@code Greeter} server.
+ */
+public class HelloWorldServer {
+  static String port_str;
+  private static final Logger logger = Logger.getLogger(HelloWorldServer.class.getName());
+
+  private Server server;
+
+  private void start(String port) throws IOException {
+    /* The port on which the server should run */
+    server = ServerBuilder.forPort(Integer.parseInt(port))
+        .addService(new GreeterImpl())
+        .build()
+        .start();
+    logger.info("Server started, listening on " + port);
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      @Override
+      public void run() {
+        // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+        System.err.println("*** shutting down gRPC server since JVM is shutting down");
+        HelloWorldServer.this.stop();
+        System.err.println("*** server shut down");
+      }
+    });
+  }
+
+  private void stop() {
+    if (server != null) {
+      server.shutdown();
+    }
+  }
+
+  /**
+   * Await termination on the main thread since the grpc library uses daemon threads.
+   */
+  private void blockUntilShutdown() throws InterruptedException {
+    if (server != null) {
+      server.awaitTermination();
+    }
+  }
+
+
+  /**
+   * Main launches the server from the command line.
+   */
+  public static void main(String[] args) throws IOException, InterruptedException {
+    /* Argument parsing */
+    if (args.length != 2) {
+      System.out.println("Usage: helloworld_server PORT zk://ADDR:PORT");
+      return;
+    }
+
+    String zk_addr;
+
+    try {
+      port_str = new String(args[0]);
+      zk_addr = new String(args[1]);
+    } catch (Exception e) {
+      System.out.println("Usage: helloworld_server PORT zk://ADDR:PORT");
+      return;
+    }
+
+    ZookeeperConnection zk_conn = new ZookeeperConnection();
+    if (!zk_conn.connect(zk_addr, "localhost", port_str)) {
+      return;
+    }
+
+    final HelloWorldServer server = new HelloWorldServer();
+    server.start(port_str);
+    server.blockUntilShutdown();
+  }
+
+  static class GreeterImpl extends GreeterGrpc.GreeterImplBase {
+
+    @Override
+    public void sayHello(HelloRequest req, StreamObserver<HelloReply> responseObserver) {
+      HelloReply reply = HelloReply.newBuilder().setMessage("Hello " + req.getName()+ " from " + port_str).build() ;
```

Comment:
Missing space after `req.getName()` and before `+`, and has extra space before semi-colon.